### ISSUE TITLE
Saving and UUID with the record in the database when UUID is provided in another system and when a Ruby statement is used.

### ISSUE THREAD

**Issue by levidamian** [neo4jrb/neo4j#1075]
Examples:

When Cypher is used a statement like this save a provided UUID in the database.
```
create (u:User{status:"Active",created_at:"2015-12-07T07:13:00.052000Z",uuid:"453cb6ef-9b99-4f7a-ac0f-cf36d5085866",updated_at:"2015-12-07T07:13:00.052000Z",name:"Book Riot"}) return u;

match (u:User) return u.name, u.uuid;
| "Book Riot" | "453cb6ef-9b99-4f7a-ac0f-cf36d5085866" |
```

However the Ruby equivalent statement, something like this fails with and error:
```
u = User.create name: 'Book Riot', uuid: '453cb6ef-9b99-4f7a-ac0f-cf36d5085866'
Neo4j::Shared::Property::UndefinedPropertyError: Undefined properties: uuid
```
But if graphaware packages like graphaware-server-all-2.3.0.35.jar and graphaware-uuid-2.3.0.35.7.jar` are used a statement like this is successful and an UUID is generated.
```
u = User.create name: 'Book Riot'
u.uuid
=> "98ebc4f2-7a9a-4f7d-8a4f-284f12a425ca"
```

**Comment by jonblack** [comment]
In my acceptance tests I want to check that the correct `uuid` is returned. The only way I can think of doing this is by setting the `uuid` manually, which results in the same error.

The [validation code here](https://github.com/neo4jrb/neo4j/blob/9b94d1fe7e9c6922c2570313827dbec8b73ea83a/lib/neo4j/shared/property.rb#L60) only checks against the attributes in the model. If you don't have an `id_property`, `uuid` is created for you, but doesn't get included in this check.

@levidamian your question can be resolved by following the [working with legacy schemas](http://neo4jrb.readthedocs.io/en/8.0.x/UniqueIDs.html#working-with-legacy-schemas) in the docs.

**Comment by cheerfulstoic** [comment]
Yep, I was going to mention #1141 as being related.  Basically the `id_property` isn't considered an attribute and so it's not changeable.
